ckanext/dcatapchharvest/harvesters_i14y.py

```diff
@@ -0,0 +1,206 @@
+import json
+
+import ckan.plugins as p
+import ckan.model as model
+from ckan.logic import NotFound, get_action
+
+from ckanext.dcat.harvesters.rdf import DCATRDFHarvester
+from ckanext.dcat.interfaces import IDCATRDFHarvester
+from ckanext.dcatapchharvest.dcat_helpers import get_pagination
+from ckanext.dcatapchharvest.harvest_helper import (
+    map_resources_to_ids,
+    check_package_change,
+    create_activity,
+)
+
+import logging
+log = logging.getLogger(__name__)
+
+
+class SwissDCATI14YRDFHarvester(DCATRDFHarvester):
+    p.implements(IDCATRDFHarvester, inherit=True)
+
+    harvest_job = None
+    current_page_url = None
+
+    def info(self):
+        return {
+            'name': 'dcat_ch_i14y_rdf',
+            'title': 'DCAT-AP Switzerland I14Y RDF Harvester',
+            'description': 'Harvester for DCAT-AP Switzerland datasets from an RDF graph desighned for i14Y'  # noqa
+        }
+
+    def validate_config(self, source_config):
+        source_config = super(SwissDCATI14YRDFHarvester, self).validate_config(source_config)  # noqa
+
+        if not source_config:
+            return source_config
+
+        source_config_obj = json.loads(source_config)
+
+        if 'excluded_dataset_identifiers' in source_config_obj:
+            excluded_dataset_identifiers = source_config_obj['excluded_dataset_identifiers']  # noqa
+            if not isinstance(excluded_dataset_identifiers, list):
+                raise ValueError('excluded_dataset_identifiers must be '
+                                 'a list of strings')
+            if not all(isinstance(item, basestring)
+                       for item in excluded_dataset_identifiers):
+                raise ValueError('excluded_dataset_identifiers must be '
+                                 'a list of strings')
+
+        if 'excluded_rights' in source_config_obj:
+            excluded_rights = source_config_obj['excluded_rights']
+            if not isinstance(excluded_rights, list):
+                raise ValueError('excluded_rights must be '
+                                 'a list of strings')
+            if not all(isinstance(item, basestring)
+                       for item in excluded_rights):
+                raise ValueError('excluded_rights must be '
+                                 'a list of strings')
+
+        return source_config
+
+    def before_download(self, url, harvest_job):
+        # save the harvest_job on the instance
+        self.harvest_job = harvest_job
+        self.current_page_url = url
+
+        return url, []
+
+    def _get_guid(self, dataset_dict, source_url=None):  # noqa
```

Comment:
Why do we need `# noqa` here?

Reply:
to long line, `# noqa` is pretty often used in this script

Reply:
we can adapt style of the `harvesters.py` separately.
is it fine for you?

Reply:
That line is 56 characters long - `# noqa` is not needed there because of line length.

It's OK for me to leave the existing code as it is, but please don't introduce overly long lines and other mess into the new code you're adding here. :)

Reply:
hehe :) this is a good idea not to bring other mess into the new code.
Will remove `# noqa` here

Reply:
hmm, interesting, when I take it away I get 
 C901 'SwissDCATRDFHarvester._get_guid' is too complex (16)..